controller/aks-cluster-config-handler.go

```diff
@@ -221,6 +221,17 @@ func (h *Handler) createCluster(config *aksv1.AKSClusterConfig) (*aksv1.AKSClust
 		return config, err
 	}
 
+	resourceClusterClient, err := aks.NewClusterClient(credentials)
+	if err != nil {
+		return config, err
+	}
+
+	logrus.Infof("Checking if cluster [%s] exists", config.Spec.ClusterName)
+
+	if aks.ExistsCluster(ctx, resourceClusterClient, &config.Spec) {
```

Comment:
I think we may want to actually update the logic in `aks.ExistsCluster`. Just by looking at it:
```
// ExistsCluster Check if AKS managed Kubernetes cluster exist
func ExistsCluster(ctx context.Context, clusterClient *containerservice.ManagedClustersClient, spec *aksv1.AKSClusterConfigSpec) bool {
	resp, err := clusterClient.Get(ctx, spec.ResourceGroup, spec.ClusterName)

	return err == nil && resp.StatusCode == 200
}
What if the user doesn't have permissions to list clusters, but the cluster exists? IMO we should either return `(bool, error)` from that function and just require that users have list permissions in our docs for AKS, or if we can't do that we should be very explicit about logging the error and possibly bubble it up. I doubt we could get a scenario where `err == nil` and the response code is up in the 500s though. I think it's very likely that it either returns 404 for both lack of permissions and object not existing, or just 403 in the former case. 
```